addons/html_editor/static/src/others/embedded_components/backend/syntax_highlighting/syntax_highlighting.js

```diff
@@ -27,28 +32,46 @@ const LANGUAGES = {
     sql: "SQL",
     diff: "Diff",
 };
-export const DEFAULT_LANGUAGE_ID = "plaintext";
+
+class CodeToolbar extends Component {
+    static template = "html_editor.CodeToolbar";
+    static props = {
+        target: { validate: (el) => el.nodeType === Node.ELEMENT_NODE },
+        getContent: { type: Function },
+        onLanguageChange: { type: Function },
+        currentLanguage: { type: String },
+    };
+    static components = { Dropdown, DropdownItem, CopyButton };
+
+    setup() {
+        super.setup();
+        this.languages = LANGUAGES;
+    }
+}
 
 export class EmbeddedSyntaxHighlightingComponent extends Component {
     static template = "html_editor.EmbeddedSyntaxHighlighting";
 
+    static components = { CodeToolbar };
     static props = {
-        initialValue: { type: String },
+        value: { type: String },
+        languageId: { type: String },
         autofocus: { type: Boolean },
-        codeToolbar: { type: Object },
         onTextareaFocus: { type: Function },
-        addHistoryStep: { type: Function },
-        getPreValue: { type: Function },
         host: { type: Object },
     };
 
     setup() {
         super.setup();
-        this.loadPrism();
+        let didAssetsLoad;
+        onWillStart(() => {
+            didAssetsLoad = this.loadPrism();
+        });
```

Comment:
Same question as in the readonly Component, is there a reason why you don't want to return `this.loadPrism()` promise so that OWL can properly wait for it to load before mounting the Component?

Reply:
Oops, same answer, didn't know that was a thing in Owl.